File: qr/qr.py
import cv2

detector = cv2.QRCodeDetector()

img = cv2.imread("./qr.png") # https://www.the-qrcode-generator.com/
retval, points, _ = detector.detectAndDecode(img)

points = points.astype(int).reshape((-1,1,2))
img = cv2.drawContours(img,[points], -1, (255,0,0), 2)
cv2.putText(img, retval, (10,15), cv2.FONT_HERSHEY_COMPLEX_SMALL,1,(0,0,255),2)
cv2.imshow("QR", img)
cv2.waitKey()

print("Detected Value: ", retval)

File: cpp/qr/qr.cpp
#include <iostream>
#include <opencv2/opencv.hpp>

using namespace cv;
int main()
{
    QRCodeDetector detector;
    Mat img = imread("./qr.png");
    std::vector<Point> points;
    std::vector<std::vector<Point> > wrapper;
    std::string value = detector.detectAndDecode(img, points);
    wrapper.push_back(points);
    drawContours(img,wrapper, -1, {255,0,0}, 2);
    putText(img, value, {10,15}, FONT_HERSHEY_COMPLEX_SMALL,1,{0,0,255},2);
    imshow("QR", img);
    waitKey();
    std::cout << "Detected value: " << value << std::endl;
}
